File: content/problems/1721-swapping-nodes-in-a-linked-list/solutions/two-pointer-value-swap/Solution.java
class Solution {
    public ListNode swapNodes(ListNode head,int k){
        ListNode front=head;
        for(int i=1;i<k;i++)front=front.next;
        ListNode back=head,runner=front;
        while(runner.next!=null){
            runner=runner.next;
            back=back.next;
        }
        int value=front.val;
        front.val=back.val;
        back.val=value;
        return head;
    }
}

File: content/problems/1721-swapping-nodes-in-a-linked-list/solutions/two-pointer-value-swap/Solution.kt
class Solution {
    fun swapNodes(head:ListNode?,k:Int):ListNode?{
        var front=head!!;
        for(i in 1 until k)front=front.next!!;
        var back=head!!;
        var runner=front;
        while(runner.next!=null){
            runner=runner.next!!;
            back=back.next!!
        };
        val value=front.`val`;
        front.`val`=back.`val`;
        back.`val`=value;
        return head
    }
}
